src/msg/async/rdma/Infiniband.h

```diff
@@ -106,6 +106,7 @@ class DeviceList {
       ceph_abort();
     }
     devices = new Device*[num];
+    ceph_assert(devices);
```

Comment:
i think the C++ runtime will throw a `bad_alloc` in this case. so no need to assert here.

Reply:
@tchaikov What's the meaning of "bad_alloc" here? I run the built out image, there's no problem yet.

Reply:
https://en.cppreference.com/w/cpp/memory/new/bad_alloc

Reply:
why do you think we would read a destroyed object?

Reply:
Previously, I don't think it's good to depend on compiler's behaviour to insert some check code into code.
Now, I'll follow your suggestion to change it.
Thanks Kefu.

Reply:
please note, it's the standard library's behavior.